spec/lib/metrics_logger_spec.rb

```diff
@@ -109,15 +111,25 @@
     subject.record_link_click(sample_visit)
     subject.record_booking_confirmation(sample_visit)
     subject.processed?(sample_visit).should be_true
-    subject.visit_status(sample_visit.visit_id).should == :confirmed
+    subject.visit_status(sample_visit.visit_id).should == 'confirmed'
   end
 
-  it "responds with a visit status as cancelled" do
-    subject.record_visit_request(sample_visit)
-    subject.record_link_click(sample_visit)
-    subject.record_booking_cancellation(sample_visit.visit_id, :visit_cancelled)
-    subject.processed?(sample_visit).should be_false
-    subject.visit_status(sample_visit.visit_id).should == :visit_cancelled
+  ['visit_cancelled', 'request_cancelled'].each do |reason|
+    it "responds with a visit status as cancelled when reason is #{reason}" do
+      subject.record_visit_request(sample_visit)
+      subject.record_link_click(sample_visit)
+      subject.record_booking_cancellation(sample_visit.visit_id, reason)
+      subject.processed?(sample_visit).should be_false
+      subject.visit_status(sample_visit.visit_id).should == reason
+    end
+  end
+
+  it "blows up when trying to cancel with an unknown reason" do
+    expect {
+      subject.record_visit_request(sample_visit)
+      subject.record_link_click(sample_visit)
+      subject.record_booking_cancellation(sample_visit.visit_id, 'whatever')
+    }.to raise_error(ActiveRecord::RecordInvalid)
```

Comment:
No, because it's not supposed to happen. It's a sanity check for when someone adds a new outcome but forgets to add it to a list of valid outcomes in the model definition.